### PR TITLE
Rename getUsers to getUsersExcludingCurrent

### PR DESCRIPTION
**Pull request by adasiewiczr**
### 🎯 Goal

Rename `getUsers` to more verbose `getUsersExcludingCurrent`

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS (task created))
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/utils/extensions/Channel.kt [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
 
-import io.getstream.chat.android.client.extensions.getUsers
+import io.getstream.chat.android.client.extensions.getUsersExcludingCurrent
 import io.getstream.chat.android.client.models.Channel
 import io.getstream.chat.android.core.internal.InternalStreamChatApi
 import io.getstream.chat.android.offline.ChatDomain
 
 @InternalStreamChatApi
-public fun Channel.isDirectMessaging(currentUserId: String = currentUserId()): Boolean = getUsers(currentUserId).size == 1
+public fun Channel.isDirectMessaging(currentUserId: String = currentUserId()): Boolean = getUsersExcludingCurrent(currentUserId).size == 1
```
Comment (zsmb13) [inline stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/utils/extensions/Channel.kt:9]: Looks like this is the only place where we pass in a `currentUserId` to the method being renamed. And then the `isDirectMessaging` function is almost never called with a parameter either (just once as far as I see, and I don't know if even that is necessary. Let's see if we can eliminate these params if they're not actually needed somewhere.

Comment (adasiewiczr) [reply stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/utils/extensions/Channel.kt:9]: We are using it inside `MessageInputView` to be sure it's initialized correctly if the view was created before a user is set. Let me check if can somehow improve that

Comment (zsmb13) [reply stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/utils/extensions/Channel.kt:9]: Maybe we can check for that inside `getUsersExcludingCurrent`, if we don't have a user, just don't filter out anything?

Comment (adasiewiczr) [reply stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/utils/extensions/Channel.kt:9]: 👍 makes sense. Then reinitialize the view if the user was set

File: stream-chat-android-client/src/main/java/io/getstream/chat/android/client/extensions/ChannelExtension.kt [hunk 2]
```diff
@@ -15,5 +15,5 @@ public fun Channel.isAnonymousChannel(): Boolean = id.isAnonymousChannelId()
 public fun Channel.isMutedFor(user: User): Boolean = user.channelMutes.any { mute -> mute.channel.cid == cid }
 
 @InternalStreamChatApi
-public fun Channel.getUsersExcludingCurrent(excludeUserId: String = ChatClient.instance().getCurrentUser()?.id ?: ""): List<User> =
-    members.map { it.user }.filterNot { it.id == excludeUserId }
+public fun Channel.getUsersExcludingCurrent(): List<User> =
+    members.map { it.user }.filterNot { it.id == ChatClient.instance().getCurrentUser()?.id ?: "" }
```
Comment (zsmb13) [inline stream-chat-android-client/src/main/java/io/getstream/chat/android/client/extensions/ChannelExtension.kt:19]: We do a bunch of comparisons to the empty string here in some cases, which is not something we want to do. How about this instead?

```suggestion
public fun Channel.getUsersExcludingCurrent(): List<User> {
    val users = members.map { it.user }

    val currentUserId = ChatClient.instance().getCurrentUser()?.id
    return if (currentUserId != null) {
        users.filterNot { it.id == currentUserId }
    } else {
        users
    }
}
```

File: stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/utils/extensions/Channel.kt [hunk 3]
```diff
@@ -1,11 +1,8 @@
...
+import io.getstream.chat.android.client.extensions.getUsersExcludingCurrent
 import io.getstream.chat.android.client.models.Channel
 import io.getstream.chat.android.core.internal.InternalStreamChatApi
-import io.getstream.chat.android.offline.ChatDomain
 
 @InternalStreamChatApi
-public fun Channel.isDirectMessaging(currentUserId: String = currentUserId()): Boolean = getUsers(currentUserId).size == 1
-
-private fun currentUserId(): String = ChatDomain.instance().user.value?.id ?: ""
+public fun Channel.isDirectMessaging(): Boolean = getUsersExcludingCurrent().size == 1
```
Comment (zsmb13) [inline stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/utils/extensions/Channel.kt:8]: It doesn't seem very efficient to create this list of users with a bunch of operations to just check for its size. How about

```suggestion
public fun Channel.isDirectMessaging(): Boolean {
    return members.size == 2 && includesCurrentUser()
}

private fun Channel.includesCurrentUser(): Boolean {
    val currentUserId = ChatClient.instance().getCurrentUser()?.id ?: return false
    return members.any { it.user.id == currentUserId }
}
```

**Review by zsmb13 — APPROVED** [review]

**Pull request merged**